### PR TITLE
[#10245] Per Question CSV Download in Session Result

### PR DESCRIPTION
**Pull request by t-cheepeng**
Part of #10245

**Screenshot**
asd

**Outline of Solution**
- Added in `downloadQuestionResult` event to base classes and `question-text-with-info`
- Added in extra `questionId` to `downloadSessionResults` to download the csv for the question only
- Bubble event upwards to root `session-result-page` and handle it there using `feedbackSessionService`
- Additional styling for question text

### CODE REVIEW

File: src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.html [hunk 1]
```diff
@@ -17,7 +17,11 @@ <h3>{{ teamInfo.key }} Statistics for {{ isGqr ? 'Given' : 'Received' }} Respons
               <div class="card-header bg-info text-white">
                 <tm-question-text-with-info [questionNumber]="question.feedbackQuestion.questionNumber"
                                             [questionDetails]="question.feedbackQuestion.questionDetails"
-                                            (click)="$event.stopPropagation()"></tm-question-text-with-info>
+                                            (click)="$event.stopPropagation()"
+                                            (downloadQuestionResultEvent)="triggerDownloadQuestionResult(
+                                            question.feedbackQuestion.questionNumber,
+                                            question.feedbackQuestion.feedbackQuestionId);"
```
Comment (Derek-Hardy) [inline src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.html:23]: Here seems to lack indentation.

File: src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.html [hunk 2]
```diff
@@ -76,7 +80,11 @@ <h3>{{ teamInfo.key }} Detailed Responses</h3>
           <div class="card-header bg-info text-white" (click)="question.isTabExpanded = !question.isTabExpanded">
             <tm-question-text-with-info [questionNumber]="question.questionOutput.feedbackQuestion.questionNumber"
                                         [questionDetails]="question.questionOutput.feedbackQuestion.questionDetails"
-                                        (click)="$event.stopPropagation()"></tm-question-text-with-info>
+                                        (click)="$event.stopPropagation()"
+                                        (downloadQuestionResultEvent)="triggerDownloadQuestionResult(
+                                        question.questionOutput.feedbackQuestion.questionNumber,
+                                        question.questionOutput.feedbackQuestion.feedbackQuestionId);"
```
Comment (Derek-Hardy) [inline src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.html:86]: Similar here

File: src/web/app/pages-instructor/instructor-session-result-page/instructor-session-result-question-view.component.html [hunk 3]
```diff
@@ -1,6 +1,10 @@
 <div class="card margin-bottom-15px" *ngFor="let question of questionsOrder">
   <div class="card-header bg-info text-white course-tab-header" (click)="toggleAndLoadTab.emit(question.question.feedbackQuestionId)">
-    <tm-question-text-with-info [questionNumber]="question.question.questionNumber" [questionDetails]="question.question.questionDetails"></tm-question-text-with-info>
+    <tm-question-text-with-info
+        [questionNumber]="question.question.questionNumber"
+        [questionDetails]="question.question.questionDetails"
+        (downloadQuestionResultEvent)="triggerDownloadQuestionResult({questionNumber: question.question.questionNumber, questionId: question.question.feedbackQuestionId})"
```
Comment (Derek-Hardy) [inline src/web/app/pages-instructor/instructor-session-result-page/instructor-session-result-question-view.component.html:6]: We can probably standardise the style

File: src/web/app/pages-instructor/instructor-session-result-page/instructor-session-result-view.ts [hunk 4]
```diff
@@ -86,4 +90,10 @@ export abstract class InstructorSessionResultView implements OnInit {
     this.saveNewCommentEvent.emit(responseId);
   }
 
+  /**
+   * Triggers the download of a question result
+   */
```
Comment (Derek-Hardy) [inline src/web/app/pages-instructor/instructor-session-result-page/instructor-session-result-view.ts:95]: Doc string should end with .

File: src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.html (hunk 2 above)
Comment (t-cheepeng) [reply src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.html:86]: The indentations follow that of the components below/above it. Maybe it appears weirdly in the git viewer?

File: src/web/app/pages-static/index-page/index-page.component.html [hunk 5]
```diff
@@ -113,7 +113,7 @@ <h2>Not for Profit:</h2>
     <h2>Well Received by the Academic Community:</h2>
     <p>
       TEAMMATES has been presented, and well-received, in education-related conferences such as the
-      <a href="http://conferences.computer.org/cseet/2011/CSEET_2011/Index.html" target="_blank" rel="noopener noreferrer">International Conference on Software Engineering Education and Training (CSEET)</a>,
+      <a href="https://conferences.computer.org/cseet/2011/CSEET_2011/" target="_blank" rel="noopener noreferrer">International Conference on Software Engineering Education and Training (CSEET)</a>,
```
Comment (wkurniawan07) [inline src/web/app/pages-static/index-page/index-page.component.html:116]: Not part of what is described in the issue title, but I will close one eye as this is still part of UAT fix

File: src/web/app/pages-instructor/instructor-session-result-page/instructor-session-result-question-view.component.html [hunk 6]
```diff
@@ -1,7 +1,13 @@
...
   <div class="card-header bg-info text-white course-tab-header" (click)="toggleAndLoadTab.emit(question.question.feedbackQuestionId)">
     <div class="row">
-      <tm-question-text-with-info class="col-md-11" [questionNumber]="question.question.questionNumber" [questionDetails]="question.question.questionDetails"></tm-question-text-with-info>
+      <tm-question-text-with-info class="col-md-11"
+        [questionNumber]="question.question.questionNumber"
+        [questionDetails]="question.question.questionDetails"
+        (downloadQuestionResultEvent)="triggerDownloadQuestionResult({
+        questionNumber: question.question.questionNumber,
+        questionId: question.question.feedbackQuestionId})"
+      ></tm-question-text-with-info>
```
Comment (wkurniawan07) [inline src/web/app/pages-instructor/instructor-session-result-page/instructor-session-result-question-view.component.html:10]: Only this needs to be changed.

File: src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.html [hunk 7]
```diff
@@ -76,7 +77,8 @@ <h3 class="top-padded">{{ teamInfo.key }} Detailed Responses</h3>
           <div class="card-header bg-info text-white" (click)="question.isTabExpanded = !question.isTabExpanded">
             <tm-question-text-with-info [questionNumber]="question.questionOutput.feedbackQuestion.questionNumber"
                                         [questionDetails]="question.questionOutput.feedbackQuestion.questionDetails"
-                                        (click)="$event.stopPropagation()"></tm-question-text-with-info>
+                                        (click)="$event.stopPropagation()"
+            ></tm-question-text-with-info>
```
Comment (wkurniawan07) [inline src/web/app/components/question-responses/gqr-rqg-view-responses/gqr-rqg-view-responses.component.html:81]: And this

File: src/web/app/components/question-responses/grouped-responses/grouped-responses.component.html [hunk 8]
```diff
@@ -27,7 +27,10 @@
     <div *ngFor="let response of responses">
       <div class="card bottom-padded">
         <div class="card-header bg-info text-white">
-          <tm-question-text-with-info [questionNumber]="response.feedbackQuestion.questionNumber" [questionDetails]="response.feedbackQuestion.questionDetails"></tm-question-text-with-info>
+          <tm-question-text-with-info
+              [questionNumber]="response.feedbackQuestion.questionNumber"
+              [questionDetails]="response.feedbackQuestion.questionDetails"
+          ></tm-question-text-with-info>
```
Comment (wkurniawan07) [inline src/web/app/components/question-responses/grouped-responses/grouped-responses.component.html:33]: And this

**Pull request merged**